java/org/apache/catalina/realm/DataSourceRealm.java

```diff
@@ -223,6 +266,34 @@ public void setUserTable( String userTable ) {
       this.userTable = userTable;
     }
 
+    /**
+     * @return the comma separated names of user attributes to additionally query
+     *         from the user table
+     */
+    public String getUserAttributes() {
```

Comment:
Why not a `String[]` or `Collection<String>`?

Reply:
That is the getter method for the Realm's `userAttributes` configuration attribute. It's configured by means of an XML attribute in the `<Realm ...>` component. It can also be read and written by through JMX. AFAIK, there should always be a getter/setter pair for every configuration property using the String type.

Reply:
I see, makes sense!